api/features_api.py

```diff
@@ -255,6 +259,30 @@ def _patch_update_special_fields(
     if should_remove_first_notice_milestone(feature, feature_changes):
       feature.first_enterprise_notification_milestone = None
 
+    # Set shipping_year based on milestones.
+    if updated_stages:
+      existing_shipping_stages = (
+          stage_helpers.get_all_shipping_stages_with_milestones(
```

Comment:
This seems like a large check for any time any stage is updated. Is it safe to check only when a shipping milestone is changed?

Reply:
Good point.  Done.